presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/UpdateablePriorityQueue.java

```diff
@@ -13,6 +13,9 @@
  */
 package com.facebook.presto.execution.resourceGroups;
 
+import javax.annotation.concurrent.ThreadSafe;
+
+@ThreadSafe
```

Comment:
Thread safety is an implementation detail, so not sure `ThreadSafe` annotation is really suitable for here. This annotation says that the annotated class is thread safe, however here you use it for saying that implementations of this interface must be thread safe, so not sure whether this is the right use for this annotation. We can just add a javadoc comment here saying implementations must be thread safe.